tests/kernel/workq/work_queue_api/src/main.c

```diff
@@ -353,7 +355,9 @@ static void twork_resubmit_nowait(void *data)
 	zassert_true(k_work_pending(&delayed_work[1].work), NULL);
 
 	/* Release sleeper, check other two still pending */
-	k_usleep(1);
+	do {
+		k_usleep(1);
+	} while (k_work_pending(&delayed_work_sleepy.work));
```

Comment:
If this is trying to assume that the the "sleepy" item is definitely going to have executed after 1us in all configurations, then that's a test bug as it's sensitive to tick rate.  You probably want k_sleep(K_TICKS(1)) here instead.

Reply:
Whether it's `k_sleep(K_TICKS(1))` or `k_usleep(1)` the first-submitted work item doesn't get processed until either 1 or 2 of these sleeps occur.  The preceding test abuses the API by reaching in and mutating the queue, so it may be that the thread is being caught in different states, one of which requires a second chance to run.